\section{Introduction}
        Molecular interaction networks (MINs) are typically represented as graphs where nodes represent proteins, nucleic acids, or metabolites and edges represent physical or functional interactions. By `functional' we refer to the indirect effect that some gene $g_i$ has on another $g_j$. None of the edges in the networks featured in this study represents a functional relationship, rather each edge represents a direct and physical interaction between  $g_i$ and  $g_j$.
        interactions. The steady increase in scale \cite{rolland_proteome-scale_2014} and resolution \cite{yang_widespread_2016} of experimentally validated interactions has not been matched with theoretical progress towards deciphering the underlying evolutionary forces shaping the structure of MINs. Earlier studies aimed to empirically show that MINs possess a certain property (say, small-world connectivity or power-law-fitted degree distribution), and subsequently advocate for the existence of a universal design principle behind it. However, the robustness of the observations and conclusions of these studies \cite{barabasi_emergence_1999, fell_small_2000} has been questioned \cite{arita_metabolic_2004, tanaka_protein_2005, fox_keller_revisiting_2005, khanin_how_2006}, and so too \cite{stelling_robustness_2004, hahn_molecular_2004} % siegal_functional_2007
        has the validity of the design principles \cite{albert_error_2000, barabasi_network_2004} they inspired. Assuming those properties were indeed real, the universality of proposed design principles around them would still need to be justified. The statistical support of a property "is no evidence of universality without a concrete underlying theory to support it" \cite{stumpf_critical_2012}. Furthermore, a universal theory "must facilitate the inclusion of domain mechanisms and details" yet there is a sharp disconnect between current hypotheses' high abstractions and actual functional aspects of evolving biological systems \cite{alderson_contrasting_2010}.
        %
        A 'software' trait relates to \textit{relationships} between genes (connectivity, community clustering, co-expression etc) rather than to their molecular ('hardware') properties (e.g. their amino acid sequence or 3D conformation).
        The assumption that natural selection can lead to the emergence of advantageous network-level software traits has itself been challenged by the non-adaptive hypothesis: % \cite{clune_evolutionary_2013}
        the topology of MINs could be an indirect result of selection pressure on other traits \cite{papp_critical_2009} or
        a mere byproduct of non-adaptive evolutionary forces such as mutation and genetic drift \cite{lynch_evolution_2007, sorrells_making_2015}. Both the adaptive and non-adaptive hypotheses present sufficient but not necessary conditions for the emergence of network traits and therefore one cannot objectively rule out the plausibility of the other.
        %The latter views effectively question  the merit of the entire scientific pursuit. There is clearly a need to ground the study of MINs particularly, and systems biology generally \cite{alderson_contrasting_2010}, in a  theoretical foundation \cite{brenner_turing_2012} that can explain and predict observed properties of biological systems. In contrast to the aforementioned approach in the field, the reverse epistemological process would likely be more fruitful: starting from a well-established universal law, could one point to a property in biological systems that must have been the result of that law imposing its constraints on their evolution?
