\documentclass[12pt]{article}	% RECOMB: "at least 11 point font size on U.S. standard 8 1/2 by 11 inch paper with no less than one inch margin all around."
%\documentclass{nature}
\usepackage[utf8]{inputenc}   % umlauts etc.
\usepackage[english]{babel}
\usepackage [autostyle, english = american]{csquotes}
\MakeOuterQuote{"}
\usepackage[hidelinks]{hyperref}
\setlength\parindent{25pt}
% ----------------------------------
%\usepackage[backend=biber,style=nature,sorting=none,url=true]{biblatex}
% url = false. There are also isbn, doi etc., similar options.
%\addbibresource{/Volumes/DataHub/Publishing/z-misc/zotero-library/better_bibtex_standalone.bib}

% ----------------------------------
% Citation style 	biblatex stylename
% ----------------------------------
% 	ACS				chem-acs
% 	AIP				phys (*)
% 	Natur			nature
% 	Science			science
% 	IEEE			ieee
% 	Chicago			chicago-authordate
% 	MLA				mla
% 	APA				apa
% ----------------------------------
% sorting options:
% ----------------------------------
%	nty 		Sort by name, title, year.
%	nyt 		Sort by name, year, title.
%	nyvt 		Sort by name, year, volume, title.
%	anyt 		Sort by alphabetic label, name, year, title.
%	anyvt 		Sort by alphabetic label, name, year, volume, title.
%	ynt 		Sort by year, name, title.
%	ydnt 		Sort by year (descending), name, title.
%	none 		Do not sort at all. All entries are processed in citation order.
% ----------------------------------
\newcommand{\harpoon}{\overset{\rightharpoonup}}
\newtheorem{theorem}{Theorem}
\usepackage{verbatim} % multiline comment
\usepackage{graphicx}
\graphicspath{{/Volumes/DataHub/Publishing/Figures/}}
\setlength\fboxsep{0pt} % figure border padding
\setlength\fboxrule{1pt} % figure outline
\usepackage[fleqn]{amsmath}  % also \documentclass[fleqn]{article}
\usepackage[margin=1in]{geometry}
%lmargin rmargin tmargin bmargin
	\geometry{
	 a4paper,
	 %total={170mm,257mm},
	 left=20mm
	 ,right=20mm
	 %,top=30mm
	 %,bottom=13mm
	 }


%\abovedisplayskip=0pt
%\abovedisplayshortskip=0pt
%\belowdisplayskip=0pt
%\belowdisplayshortskip=0pt
%
%\setlength{\mathindent}{0pt}
%
\usepackage{amsfonts} % for R (real numbers)
\usepackage{float}
% \usepackage[font=scriptsize,labelfont=bf]{caption}
\usepackage{booktabs, cellspace, hhline} % \cellspacetoplim; http://tex.stackexchange.com/questions/302960/modify-arraystretch-for-a-single-row-in-table


\usepackage[percent]{overpic}
\usepackage[export]{adjustbox}
% ----------------------------------
%Squeezing the Vertical White Space
%http://www.terminally-incoherent.com/blog/2007/09/19/latex-squeezing-the-vertical-white-space/
% 	THIS FIXES THE PROBLEM OF SUBSECTIONS STARTING IN A NEW PAGE
%   \setlength{\parskip}{5pt}
%   \setlength{\parsep}{10pt}
%   \setlength{\headsep}{0pt}
%   \setlength{\topskip}{0pt}
%   \setlength{\topmargin}{0pt}
%   \setlength{\topsep}{0pt}
%   \setlength{\partopsep}{10pt}
\usepackage[compact]{titlesec}
\titlespacing{\section}{0pt}{*2}{*2} % {left margin} {above-skip} {below-kip} , The * notation replaces the formal notation using plus/minus and etc.
\titlespacing{\subsection}{0pt}{*1}{*1}
\titlespacing{\subsubsection}{0pt}{*1}{*1}
% ----------------------------------
%\newenvironment{absolutelynopagebreak}
%  {\par\nobreak\vfil\penalty0\vfilneg
%   \vtop\bgroup}
%  {\par\xdef\tpd{\the\prevdepth}\egroup
%   \prevdepth=\tpd}
% ----------------------------------
\newcommand{\bfl}{\begin{flushleft}}
\newcommand{\efl}{\end{flushleft}}
\newcommand{\mys }{\hspace{0.1cm}}
\newcommand{\figfont}{\footnotesize}

\usepackage[table]{xcolor} % for \arrayrulecolor{yellow}, changes hline and vline colors
\definecolor{myTableLines}{HTML}{0a84f7}

\usepackage{relsize} % \tiny, \scriptsize, \footnotesize, \small, \normalsize, \large, \Large, \LARGE, \huge, \Huge
                    % if the current font size is 'normalsize', then using \relsize{-1} ==> go down one level ==> font = small
                    % and using \relsize{1}  ==> font = large
\newcommand{\myC}[1]{{\relsize{-1}{$\mathcal{#1}$}}} % as in 'my complexity class'

% padding in tables
\newcommand{\myPadTop}{6.5pt}
\newcommand{\myPadBottom}{4pt}

\mathchardef\myhyphen="2D
\usepackage{amssymb}%for QED symbol \blacksquare
\title {{\Large Computational Intractability Law Molds the Topology of Biological  Networks} \\ {\Large{\textbf{Supplementary Material}}}}

\input{/Volumes/DataHub/Publishing/Papers/Paper_01/NatSysBio/sup/tex/00_authors.tex}

%=================================================================================

\date{}

\usepackage{pifont}
\input{/Volumes/DataHub/Publishing/Papers/Paper_01/NatSysBio/sup/tex/00_authors.tex}

\begin{document}

\maketitle
\begin{comment}
\tableofcontents
{%references from main text
    \color{white}
         \cite{rolland_proteome-scale_2014, yang_widespread_2016, barabasi_emergence_1999, fell_small_2000, arita_metabolic_2004, tanaka_protein_2005, fox_keller_revisiting_2005, khanin_how_2006, stelling_robustness_2004, hahn_molecular_2004, albert_error_2000, barabasi_network_2004, stumpf_critical_2012, alderson_contrasting_2010, papp_critical_2009, lynch_evolution_2007, sorrells_making_2015, aaronson_limits_2004, aaronson_guest_2005, fortnow_status_2009, vazirani_approximation_2013, lawler_fast_1979, wigderson_opening_2014, carvunis_proto-genes_2012, vinayagam_integrating_2014, pisinger_core_1999-1, pisinger_where_2005, costanzo_global_2016, livnat_analytical_2011, chastain_algorithms_2014, livnat_sex_2016, brenner_turing_2012, gerstein_architecture_2012, khurana_interpretation_2013, kim_positive_2007, mitra_integrative_2013}
}
\end{comment}
\section{Abbreviations}
		 \begin{table}[H]
                \setlength\arrayrulewidth{.1pt}\arrayrulecolor{myTableLines}
        		\begin{tabular}[H]{r|l}
                        \textbf{NPC}    & \myC{NP}-complete \\
                        \textbf{MIN}    & Molecular interaction network \\
                		\textbf{mLmH}   & majority-Leaves minority-Hubs network topology \\
                		\textbf{OA}     & Oracle Advice \\
                		\textbf{RVnRS}  & Random variation non-random selection \\
                		\textbf{NEP}    & Network Evolution Problem \\
                		\textbf{KOP}    & Knapsack Optimization Problem \\
                		\textbf{PPI}    & Protein-protein interaction \\
                		%ER & Erdos-Renyi \\
                		\textbf{NL}     & No-Leaf network \\
                		\textbf{NH}     & No-Hub network \\
                		\textbf{$amb$}  & ambiguous \\
                		\textbf{EIS}    & Effective instance size \\
                		%\textbf{GB}     & Gained benefits {\relsize{-1}(total benefits of conserved genes in  NEP optimal solution)} \\
                		%\textbf{LB}     & Lost benefits {\relsize{-1}(total benefits of deleted genes in NEP optimal solution)} \\
                		\textbf{EGB}    & Effective gained benefits \\%(GB $\div$ (GB+LB)) \\

                        \textbf{PSICQUIC} & Proteomics Standard Initiative Common QUery InterfaCe \\
                        \textbf{MIQL} & Molecular Interaction Query Language \\
                        ${n2e}$ & node:edge ratio of a network \\
                        ${e2n}$ & edge:node ratio of a network \\
                		%UR & Unsigned and reversed direction \\
        		\end{tabular}
        \end{table}

\newpage
\section{MINs with experimental evidence and their synthetic analogs: }\label{sup_realnets}
\subsection{protein-protein interaction networks}\label{sec:data_DB_nets}
    Table \ref{tab:networks_summary_PPI}  shows the details and references of protein-protein interaction (PPI) networks (raw data and source code available in  \cite{atiia_case-study_2017}).
    PPI networks represent a "universe of possibilities", where combinatorial experiments test the affinity of each protein against all others in (typically, in large-scale experiments) exogenous settings. Widely used experimental methods include yeast two-hybrid (Y2H) and affinity purification followed by Mass spectrometry (AP-MS). Examining the literature references in Table \ref{tab:networks_summary_PPI} in chronological order of publication dates (ranging from 2008-2016), one observes a rapid increase in the scale  and resolution  of high-throughput methods with works by Rolland et \textit{al.}  \cite{rolland_proteome-scale_2014} and Yang et \textit{al.}  \cite{yang_widespread_2016} representing the cutting edge in terms of coverage and resolution respectively.
    In  \cite{yang_widespread_2016}, it was shown that different isoforms of the same protein can exhibit quite different interaction profiles. Therefore the degree of a gene (particularly hub genes) may in fact be inflated in networks where isoforms are not distinguished: that gene should ideally be broken down to separate nodes corresponding to each isoform.
    Typically, further validation of the resulting networks is conducted on a subset of interactions by testing their affinity in endogenous setting (which in turn is used to calculate some measure of true/false positives/negatives or some combination of such ratios) or comparing the resulting interactions to (small) gold standard data sets. It is important to note that PPI networks are generally undirected, since the experimental methods only establish the existence of an interaction but reveal nothing about the type (whether promotional or inhibitory) or directionality (which of the two proteins affects the other) of an interaction. The Fly network is the one exception, as both the direction and type of its interactions have been assessed using a simple prediction algorithm which achieved "90\% precision and 41\% recall (2.8\% false positive rate and 59\% false negative rate)"  \cite{vinayagam_integrating_2014}.
    %All of PPI networks in this study were mapped using the yeast two-hybrid (Y2H) experimental method with the exception of Fly which was generated using RNA interference (RNAi) screens.
    Figure SI \ref{fig:deg_dist_PPI} shows the degree distribution of PPI networks and their corresponding synthetic analogs which were generated using the same method discussed in Section "Simulation of evolutionary pressure" in the main text.
    %\setlength{\textfloatsep}{0pt plus 1.0pt minus 1.0pt} % http://tex.stackexchange.com/questions/26521/how-to-change-the-spacing-between-figures-tables-and-text
    \newpage
    \begin{table}[H]%[!htb] %h:here, t:top of page, b:bottom of page, more: http://tex.stackexchange.com/questions/35125/how-to-use-the-placement-options-t-h-with-figures
        \centering
		\setlength\arrayrulewidth{.1pt}\arrayrulecolor{myTableLines}%\arrayrulecolor[HTML]{0a84f7} %https://en.wikibooks.org/wiki/LaTeX/Colors#Adding_the_color_package
		\scriptsize %\small \tiny, \scriptsize, \footnotesize, \small, \normalsize, \large, \Large, \LARGE, \huge, and \Huge.
			\setlength\cellspacetoplimit{\myPadTop} % padding in table cells
			\setlength\cellspacebottomlimit{\myPadBottom} %padding in table cells
			\begin{tabular}{@{}Sc Sc|Sc|Sc|Sc|Sc|Sc@{}}  % http://tex.stackexchange.com/questions/302960/modify-arraystretch-for-a-single-row-in-table
				\cline{2-7}
                    %
					& \textbf{\normalsize PPI Network} & \textbf{\normalsize no. nodes } & \textbf{\normalsize no. edges }	& \textbf{\normalsize e2n ratio } & \textbf{\normalsize directed? } & \textbf{\normalsize signed? }
                    %
                %\\[.05cm] \cline{1-6} % \multirow{''num_rows''}{''width''}{''contents''}
				%	\multirow{7}{*}{\textbf{\normalsize PPIs}} & Plant  \cite{consortium_evidence_2011}  &  2661  &  5664  &  2.13  &   0.450
                        \\[.05cm] \cline{2-7}\cline{2-7}\cline{2-7}
                            & Plant  \cite{consortium_evidence_2011}  &  2661  &  5664  &  2.13  & no & no %&   0.450
        				\\[.05cm] \cline{2-7}\cline{2-7}\cline{2-7}
        					& Bacteria  \cite{rajagopala_binary_2014} &  1267  &  2233   &  1.76  & no & no %&   0.567
        				\\[.05cm] \cline{2-7}
        					& Yeast  \cite{yu_high-quality_2008}      &  2018  &  2930   &  1.45  & no & no %&   0.689
        				\\[.05cm] \cline{2-7}
        					& Worm  \cite{simonis_empirically_2009}   &  2528  &  3864   &  1.53  & no & no %&   0.654
        				\\[.05cm] \cline{2-7}
        					& Fly  \cite{vinayagam_integrating_2014}  &  3352  &  6094   &  1.82  & yes & yes %&   0.550
        				\\[.05cm] \cline{2-7}
        					& Human  \cite{rolland_proteome-scale_2014}      &  4303  &  13944  &  3.24  & no & no %&   0.309
        				\\[.05cm] \cline{2-7}
        					& HumanIso  \cite{yang_widespread_2016}   &  629  &  996     &  1.58  & no & no %&   0.632
                        %\\[.05cm] \cline{1-6}
                        \\[.05cm] \cline{2-7}
            \end{tabular}
            \caption[Summary of PPI networks.]
                    {
                        Summary of protein-protein interaction (PPI) networks. The direction and sign of an interaction were assigned at random (coin flip) in undirected and/or unsigned networks. References, data and source code publicly available in  \cite{atiia_case-study_2017}.
                     }
            \label{tab:networks_summary_PPI}
    \end{table}

    \begin{figure}[H]%[!htb]
        \includegraphics[width=.49\textwidth]{02.degree-dist/Thesis/PPIs/PPI.png}
        \includegraphics[width=.49\textwidth]{02.degree-dist/Thesis/PPIs/NH.png}
        \\
        \includegraphics[width=.49\textwidth]{02.degree-dist/Thesis/PPIs/NL.png}
        \includegraphics[width=.49\textwidth]{02.degree-dist/Thesis/PPIs/RN.png}
        \caption
                [
                    Degree distribution of PPI networks.
                ]
                {
                    Degree distribution of PPI networks and their corresponding synthetic analogs: no-hubs (NH), no-leaves (NL) and random (RN).
                }
        \label{fig:deg_dist_PPI}
    \end{figure}

\newpage
\subsection{Regulatory networks}\label{sup_realnets_reg}
    Regulatory networks (details and references in Table \ref{tab:networks_summary_Reg}, raw data and source code available in  \cite{atiia_case-study_2017}) are all directed, with some being partially signed (RegulonDB and TRRUST). The nodes in regulatory networks can be transcription factors, genes (which can refer to the protein or mRNA), or small RNAs. All networks originally contain exclusively experimentally-validated interactions, with the exception of Liu and RegulonDB which contain computationally (\textit{in-silico}) inferred interactions which were excluded. In the case of RegulonDB, only interactions with  `strong' or `confirmed' experimental evidence are included, and since none of the interactions involving small RNAs had such evidence, they were eliminated. The remaining interactions were therefore exclusively between transcription factors. In miRTarBase networks, only interactions with strong experimental evidence (elucidated through reporter assays or western blot experiments) are included. Furthermore, interactions where the species of source and target genes are different were excluded (presumably, these original from  transgenic studies).

    The ENCODE proximal network is an overall consolidated network of transcriptional interactions in humans, with some interactions being obtained by further consolidation with PPI network (detailed in supplementary materials of  \cite{gerstein_architecture_2012}). The other two ENCODE networks on the other hand are generated from specific human cell lines (GM and K562). The TRRUST network is unique in that it was obtained by data mining ${\sim}$20 million literature abstracts from  Medline (2014), out of which ${\sim}$23K sentences were nominated to contain potential descriptions of regulatory interactions  \cite{han_trrust:_2015}. These sentences underwent successive rounds of manual inspections. TRRUST network also includes information about the nature of interactions and the number of studies supporting it. For interactions deemed promotional by some studies and inhibitory by others, we picked the sign randomly by flipping a crooked coin proportional to the number of studies that support one type or another (for example, if 3 studies report an interaction as `promotional' and 1 reports it as `inhibitory', we would consider the interaction to be `promotional' with 75\% likelihood). TRRUST authors aimed to create a high-quality network that can serve as a gold-standard to other large-scale studies aiming to map transcriptome interactions in humans. The same crooked coin strategy was used in RegulonDB network. Figure SI \ref{fig:deg_dist_Reg} shows the degree distributions of regulatory networks and their corresponding synthetic analogs. Despite the diverse methods that were behind the mapping of these networks (in contrast to PPIs, where Y2H method is dominant), the mLmH property still holds with lower-degree nodes in particularly being of almost the same frequency in the majority of networks.
    \newpage
    \begin{table}[H]%[!htb]
        \centering
        \setlength\arrayrulewidth{.1pt}\arrayrulecolor{myTableLines}%\arrayrulecolor[HTML]{0a84f7}
        \scriptsize
            \setlength\cellspacetoplimit{\myPadTop}\setlength\cellspacebottomlimit{\myPadBottom}
            \begin{tabular}{@{}Sc Sc|Sc|Sc|Sc|Sc|Sc@{}}
                \cline{2-7}
                    & \textbf{\normalsize Regulatory Network} & \textbf{\normalsize no. nodes} & \textbf{\normalsize no. edges}	& \textbf{\normalsize e2n ratio} & \textbf{\normalsize directed? } & \textbf{\normalsize signed? }%&  \textbf{\normalsize n2e ratio}
                    %\\[.05cm] \cline{1-6}
    				%    \multirow{10}{*}{\textbf{\normalsize Regulatory}} & TRRUST  \cite{han_trrust:_2015}   &  2718  &  8015  &  2.95  &   0.340
                            \\[.05cm] \cline{2-7}
                                    & Bacteria RegulonDB  \cite{gama-castro_regulondb_2016} & 898       & 1481      & 1.649    & yes & no % & 0.606
                            \\[.05cm] \cline{2-7}
                                    &  ENCODE Proximal  \cite{gerstein_architecture_2012}   & 9057      & 26070     & 2.878    & yes & no %& 0.347
                            \\[.05cm] \cline{2-7}
                                    &  ENCODE K562  \cite{gerstein_architecture_2012}       & 3947      & 9595      & 2.431    & yes & no % & 0.411
                            \\[.05cm] \cline{2-7}
                                    &  ENCODE GM  \cite{gerstein_architecture_2012}         & 3989      & 6971      & 1.748    & yes & no % & 0.572
                            \\[.05cm] \cline{2-7}
                                    & Human Liu  \cite{liu_regnetwork:_2015}                & 3502      & 9606      & 2.743    & yes & no %& 0.365
                            \\[.05cm] \cline{2-7}
                                    & Human TRRUST  \cite{han_trrust:_2015}                 & 2718      & 8015      & 2.949    & yes & yes %& 0.339
                            \\[.05cm] \cline{2-7}
                                    & Human miRTarBase  \cite{chou_mirtarbase_2016}         & 2583      & 5450      & 2.11     & yes & no %& 0.474
                            \\[.05cm] \cline{2-7}
                                    & Mouse Liu  \cite{liu_regnetwork:_2015}                & 1436      & 3673      & 2.558    & yes & no % & 0.391
                            \\[.05cm] \cline{2-7}
                                    & Mouse miRTarBase  \cite{chou_mirtarbase_2016}         & 741       & 1019      & 1.375    & yes & no %& 0.727
                            \\[.05cm] \cline{2-7}
                            %\\[.05cm] \cline{1-6}
            \end{tabular}
            \caption[Summary of regulatory networks.]
                    {
                        Summary of regulatory networks. The direction and sign of an interaction were assigned at random (coin flip) in undirected and/or unsigned networks. References, data and source code publicly available in  \cite{atiia_case-study_2017}.
                    }
            \label{tab:networks_summary_Reg}
        \end{table}


        \begin{figure}[H]%[!htb]
            \includegraphics[width=.49\textwidth]{02.degree-dist/Thesis/Regulatory/Reg.png}
            \includegraphics[width=.49\textwidth]{02.degree-dist/Thesis/Regulatory/NH.png}
            \\
            \includegraphics[width=.49\textwidth]{02.degree-dist/Thesis/Regulatory/NL.png}
            \includegraphics[width=.49\textwidth]{02.degree-dist/Thesis/Regulatory/RN.png}
            \caption
                    [
                        Degree distribution of regulatory networks.
                    ]
                    {
                        Degree distribution of regulatory networks and their corresponding synthetic analogs: no-hubs (NH), no-leaves (NL) and random (RN).
                    }
            \label{fig:deg_dist_Reg}
        \end{figure}

\newpage
\subsection{Database-sourced networks}\label{sup_realnets_DB}
Table \ref{tab:networks_summary_DB} shows details and source references of networks obtained from the  BioGrid database or from multiple databases queried simultaneously through the PSICQUIC web service (raw data and source code available in  \cite{atiia_case-study_2017}).
All obtained interactions are undirected and unsigned. Interactions in BioGrid networks represent physical interactions which have been validated by at least two studies, except for human and yeast networks in which interactions have been validated by at least 4 and 3 studies, respectively (because of the large number of interactions for these two species, it was still possible to obtain large networks even under this stringent selection criteria). Multiple databases  (excluding BioGrid) were searched programmatically with a Molecular Interaction Query Language (MIQL) query through the  PSICQUIC web service interface (source code publicly available in  \cite{atiia_case-study_2017}).
The query specifies interactions where both interactors (1) are from the same species,  (2) they interact physically, and (3) the interaction has been experimentally detected.
It should be noted that some PSICQUIC interactions did distinguish whether an interactor is an isoform of a well-known gene. Figure SI \ref{fig:deg_dist_DB} shows the degree distribution of the resulting networks and their corresponding synthetic analogs. The Plant-PSICQUIC network is anomalous in its degree distribution, indicating sporadic coverage of its reported interactions. Other networks of even smaller size still exhibit the mLmH property, which can be a sign that the underlying studies behind them were less sporadic in their coverage (i.e. focusing on specific functional units).


\newpage

%#########################################################################################################################
            \begin{table}[H]%[H]%[!htb]
                \centering
                \setlength\arrayrulewidth{.1pt}\arrayrulecolor{myTableLines}%\arrayrulecolor[HTML]{0a84f7}
                \scriptsize
                    \setlength\cellspacetoplimit{\myPadTop}\setlength\cellspacebottomlimit{\myPadBottom}
                    \begin{tabular}{@{}Sc Sc|Sc|Sc|Sc|Sc|Sc@{}}
                    \cline{2-7}
                        & \textbf{\normalsize DB-sourced Network} & \textbf{\normalsize no. nodes} & \textbf{\normalsize no. edges}	& \textbf{\normalsize e2n ratio} & \textbf{\normalsize directed? } & \textbf{\normalsize signed? } %&  \textbf{\normalsize n2e ratio}
                        %\\[.05cm] \cline{2-7}
                        %\multirow{9}{*}{\textbf{\normalsize DB-sourced}} & Human-BioGrid  \cite{chatr-aryamontri_biogrid_2017}  & 3436      & 8254      & 2.402     & 0.416
                            \\[.05cm] \cline{2-7}
                                    & Plant-BioGrid  \cite{chatr-aryamontri_biogrid_2017}  & 1565      & 2745      & no & no %& 1.754     & 0.57
                            \\[.05cm] \cline{2-7}
                                   & Plant-PSICQUIC  \cite{aranda_psicquic_2011}         & 230       & 789       & 3.43   & no & no   % & 0.292
                            \\[.05cm] \cline{2-7}
                                    & Yeast-BioGrid  \cite{chatr-aryamontri_biogrid_2017}  & 2418      & 7668      & 3.171  & no & no    %& 0.315
                            \\[.05cm] \cline{2-7}
                                    & Yeast-PSICQUIC  \cite{aranda_psicquic_2011}        & 767       & 1386      & 1.807  & no & no    %& 0.553
                            \\[.05cm] \cline{2-7}
                                    & Worm-BioGrid  \cite{chatr-aryamontri_biogrid_2017}   & 55        & 64        & 1.164  & no & no    %& 0.859
                            \\[.05cm] \cline{2-7}
                                    & Fly-BioGrid  \cite{chatr-aryamontri_biogrid_2017}    & 188       & 279       & 1.484  & no & no    %& 0.674
                            \\[.05cm] \cline{2-7}
                                    & Mouse-BioGrid  \cite{chatr-aryamontri_biogrid_2017}  & 1031      & 1497      & 1.452  & no & no    %& 0.689
                            \\[.05cm] \cline{2-7}
                                    & Human-BioGrid  \cite{chatr-aryamontri_biogrid_2017}  & 3436      & 8254      & 2.402  & no & no    %& 0.416
                            \\[.05cm] \cline{2-7}
                                    & Human-PSICQUIC  \cite{aranda_psicquic_2011}        & 3470      & 6188      & 1.783  & no & no    %& 0.561
                            \\[.05cm] \cline{2-7}
                            %\\[.05cm] \cline{1-6}
                    \end{tabular}
                    \caption[Summary of database-sourced networks.]
                            {
                                Summary of real database-sourced networks. The direction and sign of an interaction were assigned at random (coin flip) in undirected and/or unsigned networks. References, data and source code publicly available in  \cite{atiia_case-study_2017}.
                            }
                    \label{tab:networks_summary_DB}
        	\end{table}

            \begin{figure}[H]%[!htb]
                \includegraphics[width=.49\textwidth]{02.degree-dist/Thesis/DB_sourced/DB.png}
                \includegraphics[width=.49\textwidth]{02.degree-dist/Thesis/DB_sourced/NH.png}
                \\
                \includegraphics[width=.49\textwidth]{02.degree-dist/Thesis/DB_sourced/NL.png}
                \includegraphics[width=.49\textwidth]{02.degree-dist/Thesis/DB_sourced/RN.png}
                \caption
                        [
                            Degree distribution of DB-sourced networks.
                        ]
                        {
                            Degree distribution of DB-sourced networks  and their corresponding synthetic analogs: no-hubs (NH), no-leaves (NL) and random (RN).
                        }
                \label{fig:deg_dist_DB}
            \end{figure}



\newpage
\section{Formal Definition of the Network Evolution Problem (NEP)} \label{sup_NEP_definition}
				\noindent Given: %http://tex.stackexchange.com/questions/82240/unwanted-space-before-flalign
				\vspace{.3cm}
				{\small
				%\setlength{\abovedisplayskip}{0pt}\setlength{\belowdisplayskip}{0pt}\setlength{\abovedisplayshortskip}{0pt}\setlength{\belowdisplayshortskip}{0pt}
						\begin{flalign*} % {align} produces equation numbers
							&\mys\mys\boldsymbol{G}  = (g_1,g_2,\dots,g_n) \textrm{, }  \boldsymbol {A}  = (a_1,a_2,\dots,a_n)  \textrm{, }  a_j \in \{+1,0,-1\} \textrm{, } \mys\boldsymbol {t}\in\mathbb{R}\textrm{, \mys and }  \\
							&\mys\mys\boldsymbol{M}  = \big [m_{jk}\big ] \quad \text{where} \quad  m_{jk} \in  \mathbb{R}, \mys \quad\forall j,k, \mys 1\leq j,k \leq n %\mathbb{R},
											%\begin{bmatrix}%nside math mode, in order to use a different (smaller) font, you could/should use \scriptstyle or \scriptscriptstyle
											%				m_{11} & m_{12} & \dots  & m_{1n} \\
											%				m_{21} & m_{22} & \dots  & m_{2n} \\
											%				\vdots & \vdots & \ddots & \vdots \\
											%				m_{n1} & m_{n2} & \dots & m_{nn}
											%			\end{bmatrix}
											%	\quad \text{where} \quad  m_{jk} \in  \{+1,0,-1\} & %\mathbb{R},
						\end{flalign*}
				}

				\noindent Let:
				{\small
				%\setlength{\abovedisplayskip}{0pt}\setlength{\belowdisplayskip}{0pt}\setlength{\abovedisplayshortskip}{0pt}\setlength{\belowdisplayshortskip}{0pt}
						\begin{flalign*}
							 \mys\mys\boldsymbol {B}  = (b_1,b_2,\dots, b_n) \textrm{,}\hspace{0.2cm}&\textrm{where}\hspace{0.2cm} b_j = \sum\limits_{k=1}^{n} m_{jk} \oplus a_k \hspace{0.15cm}+\hspace{0.15cm}\sum\limits_{k=1}^{n} m_{kj} \oplus a_j \hspace{.26cm}\textrm{and} \hspace{.4cm} &\\
								 &  m_{xy} \oplus a_y  =
											\scriptscriptstyle{\begin{cases}	% in math mode, use scriptstyle/scriptscriptstyle, not small/tiny
													|m_{xy}| & \mys\textrm{if}\quad m_{xy} \times a_y >0 \\
													\mys\mys 0 & \mys\textrm{otherwise}
											\end{cases}
											} &
						\end{flalign*}
				}
				{\small
						%\setlength{\abovedisplayskip}{0pt}\setlength{\belowdisplayskip}{0pt}\setlength{\abovedisplayshortskip}{0pt}\setlength{\belowdisplayshortskip}{0pt}
						\begin{flalign*}
						\mys\mys\boldsymbol {D}  = (d_1,d_2,\dots, d_n) \textrm{,}\hspace{0.15cm}&\textrm{where}\hspace{0.2cm} d_j = \sum\limits_{k=1}^{n} m_{jk} \ominus a_k \hspace{0.15cm}+\hspace{0.15cm} \sum\limits_{k=1}^{n} m_{kj}\ominus a_j \hspace{.21cm}\textrm{and} \hspace{.4cm} &\\
								 & m_{xy} \ominus a_y =
											\scriptscriptstyle{\begin{cases}
															|m_{xy}| & \mys\textrm{if}\quad m_{xy} \times a_y < 0 \\
															\mys\mys 0 & \mys\textrm{otherwise}
											\end{cases}
											} &
						\end{flalign*}
					}

				\noindent Define:
					{\small
					  %\setlength{\abovedisplayskip}{0pt}\setlength{\belowdisplayskip}{0pt}\setlength{\abovedisplayshortskip}{0pt}\setlength{\belowdisplayshortskip}{0pt}
						\begin{flalign*}
							&\mys\mys f:\boldsymbol{G}  \rightarrow \{0,1\} \mys \textrm{maximizing} \mys  \sum\limits_{j=1}^{n} f(g_j)\times b_j
							\mys\mys\textrm{s.t.} \mys\mys \Bigg(  \sum\limits_{j=1}^{n} f(g_j)\times d_j  \Bigg)  \leq \boldsymbol{t} &
						\end{flalign*}
					}

Table \ref{informal_table} provides a summary of each element of NEP and its corresponding semantic interpretation in biological context (see also the main text for more on the semantics of NEP in biological context).

\begin{comment}
	The network can equivalently be represented as an adjacency matrix $M$, whereby a non-zero entry $m_{jk}$ indicates the existence of an interaction between genes $g_j$ and $g_k$
	in which the latter is the target of the former.
	The sign of a non-zero entry in $M$ indicates whether $g_j$'s effect on its target $g_k$ is promotional or inhibitory in nature, indicated with
	$+1$ or $-1$, respectively.
	A hypothetical Oracle advice (OA) on all or some of the genes simulates the evolutionary pressure on the
	network, and
	is represented as a ternary sequence $A = (a_1,a_2,\dots,a_n)$ where: $a_j=+1$ ($a_j=-1$) implies
	the organism would be better off conserving (deleting) $g_j$; $a_j=0$ implies the Oracle has no opinion on $g_j$.
	While $m_{jk}$ describes what the effect of $g_j$ on $g_k$ actually \textit{is}, $a_k$ describes whether that effect \textit{should} ideally be.
	An interaction $m_{jk}$
	is beneficial if it is in agreement with what the Oracle
	says $g_k$ should be (i.e. either $(m_{jk}=+1$ AND $a_k=+1)$ OR $(m_{jk}=-1$ AND $a_k=-1)$), and damaging if it is
	in disagreement with what the Oracle says $g_k$ should be (i.e. either $(m_{jk}=+1$ AND $a_k=-1)$ OR $(m_{jk}=-1$ AND $a_k=+1)$).
	Each gene $g_j$ is henceforth assigned a benefit (damage) score $b_j$ ($d_j$) depending on how many beneficial (damaging)
	interactions it \textit{projects} onto or \textit{attracts} from other genes through its
	outgoing and incoming edges, respectively.
	Each beneficial (damaging) interaction therefore adds $|m_{jk}|$ to the benefit (damage) score of both the source gene $g_j$ and the target gene $g_k$.
	A gene can therefore have both non-zero benefit and damage score under a given pressure scenario, and so the optimization problem is:
	what subset of genes should be conserved and which should be deleted (=define $f$) so as to maximize (minimize) the number of
	interactions that are in agreement (disagreement) with the OA?
	The OA can be imposed by conserving (deleting) every gene $\boldsymbol{g_j}$ where $a_j = +1$ ($a_j=-1$).
	However,
	conserving $\boldsymbol{g_j}$ can inadvertently contribute to a violation of  the OA if
	$g_j$ happens to be a promoter (inhibitor) of one or more $g_k$ where $a_k=-1$($a_k=+1$), and
	deleting $\boldsymbol{g_j}$ can inadvertently contribute to a violation of the OA if $g_j$
	happens to be a promoter (repressor) of one or more $g_k$ where $a_k=+1$ ($a_k=-1$).
	The idealistic pursuit of enforcing an OA is complicated by the reality of network connectivity.
\end{comment}


%\setlength{\textfloatsep}{0pt plus 1.0pt minus 1.0pt} % http://tex.stackexchange.com/questions/26521/how-to-change-the-spacing-between-figures-tables-and-text
	\begin{table}[H]%[t] %h:here, t:top of page, b:bottom of page, more: http://tex.stackexchange.com/questions/35125/how-to-use-the-placement-options-t-h-with-figures
		\setlength\arrayrulewidth{.1pt}\arrayrulecolor[HTML]{0a84f7} %https://en.wikibooks.org/wiki/LaTeX/Colors#Adding_the_color_package
        {\footnotesize % \tiny, \scriptsize, \footnotesize, \small, \normalsize, \large, \Large, \LARGE, \huge, and \Huge.
			\setlength\cellspacetoplimit{4pt} % padding in table cells
			\setlength\cellspacebottomlimit{4pt} %padding in table cells
			\begin{tabular}{ @{}c|Sl@{}}
                % http://tex.stackexchange.com/questions/302960/modify-arraystretch-for-a-single-row-in-table
				\hline
					$\boldsymbol{G}  = (g_1,g_2,\dots,g_n)$
					&
					A sequence of \textbf{\textit{G}}enes: any transcribable element on the genome
				%---------------------------------------------------------------------------------------------
				\\[.01cm] \hline
					\shortstack[c]{ $\boldsymbol {A}  = (a_1,a_2,\dots,a_n)$ \\ \mys }
					&
					\shortstack[l]{\\A ternary string representing an Oracle \textbf{\textit{A}}dvice:  \\
										$a_j =
											\begin{cases}
													+1 \quad & \implies g_j \mys \text{is advantageous}
													\\
													-1 \quad & \implies g_j  \mys \text{is disadvantageous}
													\\
													0 \quad & \implies \text{no opinion on} \mys  g_j
												\end{cases}	 $
						}
				\\[.01cm] \hline
				%---------------------------------------------------------------------------------------------
						$\boldsymbol  {M} = \big [m_{jk}\big ] $
							%	\scriptscriptstyle{\begin{bmatrix}
							%		m_{11} & m_{12} & \dots  & m_{1n} \\
							%		\vdots & \vdots & \ddots & \vdots \\
							%		m_{n1} & m_{n2} & \dots & m_{nn}
							%	\end{bmatrix}}$
						&
						\shortstack[l]{ $n\times n$ \mys \textbf{\textit{I}}nteraction \textbf{\textit{M}}atrix:
								$m_{jk} =
									\begin{cases}
										p>0 \quad &\implies g_j \mys  \text{promotes} \mys   g_k
										\\
										q<0 \quad &\implies g_j \mys  \text{represses} \mys   g_k
										\\
										0 \quad\quad & \implies  g_j \mys  \text{and} \mys   g_k  \mys  \text{don't interact}
									\end{cases}$
						}
				\\[.01cm] \hline
				%---------------------------------------------------------------------------------------------
					$\boldsymbol {t} \in\mathbb{R}$
					&
					\textit{\textbf{t}}olerance threshold on damages%, expressed as a \% of total damage network-wide.  %  $\Bigg(  \sum\limits_{j=1}^{n} f(g_j)\times d_j  \Bigg)$
				\\[.01cm] \hline
				%---------------------------------------------------------------------------------------------
					% \shortstack allows for line break within the cell
					 \shortstack[c]{$\boldsymbol{B} = (b_1,b_2, \dots, b_n)$ \\  $\boldsymbol{D} = (d_1,d_2, \dots, d_n)$}
					&
					\shortstack[l]{Each gene   $g_j$   has a corresponding benefit value   $\boldsymbol{b_j}$   and damage value   $\boldsymbol{d_j}$ \\ given an Oracle advice on $g_j$ and
					all or some of its interaction partners. }
				\\[0.01cm] \hline
				%---------------------------------------------------------------------------------------------
				\shortstack[c]{\\$m_{xy} \oplus a_y  =$\tiny{$
								\begin{cases}
									|m_{xy}| & \textrm{if}\mys m_{xy}\times a_y >0 \\
									\mys\mys 0 & \textrm{otherwise}
								\end{cases}	$}}
						&
							%\shortstack[l]{ \\
							\parbox[t]{.7\textwidth}{\raggedright If the effect of  $g_x$   on   $g_y$  is in \textbf{agreement} with what the Oracle
							says   $g_y$   should be (i.e.  $m_{xy}$   and
							$a_y$  have the \newline same sign),  then increment   $b_x$  by $|m_{xy}|$}
				\\[-0.1cm] \hline
				%---------------------------------------------------------------------------------------------
				\shortstack[c]{\\$m_{xy} \ominus a_y  =$\tiny{$
								\begin{cases}
									|m_{xy}| & \textrm{if}\mys m_{xy}\times a_y < 0 \\
									\mys\mys 0 & \textrm{otherwise}
								\end{cases}	$}}
							&
							%\shortstack[l]{ \\
							\parbox[t]{.7\textwidth}{\raggedright If the effect of  $g_x$   on   $g_y$  is in \textbf{disagreement} with what the Oracle
							says   $g_y$   should be (i.e.  $m_{xy}$   and
							$a_y$  have different signs), then increment   $d_x$ by $|m_{xy}|$}
				\\[-0.1cm] \hline
				%---------------------------------------------------------------------------------------------
					\parbox[t]{.2\textwidth}{\centering
					%\begin{tabular}{c}
								$ f:\boldsymbol{G}  \rightarrow \{0,1\}$   %\\

								maximizing:

								$\sum\limits_{j=1}^{n} f(g_j)\times b_j$ %\\

								subject to:

								$\Bigg(  \sum\limits_{j=1}^{n} f(g_j)\times d_j  \Bigg)  \leq \boldsymbol{t}$
						%\end{tabular}
						}
						&
						\parbox[t]{.72\textwidth}{\raggedright %\parbox[position][height][inner-pos]{width}{text}
						%\begin{tabular}[H]{m{.7\textwidth}}
							The idealistic pursuit of enforcing an Oracle advice (OA) is complicated by the reality of network connectivity:\newline
							OA can be imposed by deleting every gene $\boldsymbol{g_i}$ where $a_i = -1$  and conserving every gene $\boldsymbol{g_j}$
							where $a_j = +1$. However: \textbf{deleting} $\boldsymbol{g_i}$ can inadvertently contribute to a violation of the OA if $g_i$
							happens to be a promoter (repressor) of some $g_k$ that should in fact be promoted  (repressed); and \textbf{conserving} $\boldsymbol{g_j}$
							can inadvertently contribute to a violation of the OA if $g_j$ happens to be a promoter (repressor) of some $g_k$ that should in fact be
							repressed (promoted). What subset of genes should be conserved/deleted (define $f$) such that the OA is supported by as many interactions
							as possible (the $maximize$ .. $subject$ $to$.. clauses)? }
					\\[.01cm] \hline


			\end{tabular}
            }%end small size
        \caption{ The syntax (left column) and semantics (right) of the network evolution problem (NEP)}
        \label{informal_table}
        \end{table}

\newpage
\section{NP-hardness of NEP}\label{KOPdefinition}
    The $\mathcal{NP}\myhyphen$hard knapsack optimization problem (KOP)  \cite{karp_reducibility_2010} is defined as:
    Given a sequence of objects $\boldsymbol{O}=(o_1,o_2,\dots,o_r), \mys$ values $\boldsymbol{V}=(v_1,v_2,\dots,v_r)$,
    weights $\boldsymbol{W}=(w_1,w_2,\dots,w_r)$, and a knapsack capacity $\boldsymbol{c}$ where $v_i, w_i, \boldsymbol{c}  \in \mathbb{N}$,
    define:

    $f:\boldsymbol{O}  \rightarrow \{0,1\}$ maximizing $\sum\limits_{j=1}^{r} f(o_j)\times v_j$ s.t. $ \Bigg(  \sum\limits_{j=1}^{r} f(o_j)\times w_j  \Bigg)  \leq \boldsymbol{c}$.

    \vspace{20pt}
    \noindent\textbf{Theorem}: \textit{NEP is $\mathcal{NP}\myhyphen$hard by reduction from KOP}.
    \vspace{20pt}


\subsection{Proof sketch}\label{sup_proof_sketch}
				For a given KOP instance with $r$ items, create a graph with $r+1$ nodes: $n_1, n_2\dots, n_{r+1}$. Assume an OA where $a_i=+1 \mys \forall a_i \in  A$ except for $a_{r+1}=0$.
				For each $v_i\in\boldsymbol{V}$, draw a $v_i$-weighted edge from $n_i$ to itself.
				Sort objects in $\boldsymbol{O}$ ascendingly by their respective weights in $\boldsymbol{W}$, call this sorted
				list $\boldsymbol{O}^{\prime}$.
				$\forall w_i\in\boldsymbol{W}$, draw a $-w_i$-weighted edge from $n_i$ to $n_j$ where $o_j$ is
				the successor of $o_i$ in $\boldsymbol{O}^{\prime}$.
				Because $n_j$ is attracting damaging interactions due to incoming edges from $n_i$,
				update its weight to $w_j-w_i$.
				For the last node $n_r$, draw $-w_r$-weighted edge from node $n_{r+1}$ to $n_r$.
				Because $n_{r+1}$ has zero-value, it's ruled out \textit{a priori} from the solution vector.

%\section{Attraction-only simulations}

%\section{Projection-only simulations}

		\subsection{Proof}\label{sup_proof}

				%\noindent$\normalsize\boldsymbol{Proof:}$

				\vspace{5pt}
				\small{\noindent I. Define $\boldsymbol{\gamma}:\{1,..,r\}\rightarrow \{1,..,r\}$ \mys s.t. $\forall i, 1\leq i <r: w_{\gamma(i)} \leq w_{\gamma(i+1)}$

							\noindent II. Let $\boldsymbol{G}=\boldsymbol{O}+\{o_{r+1}\}$, \mys $\boldsymbol{t}=\boldsymbol{c}$,\mys $\boldsymbol{A}=(a_1,\dots,a_r, a_{r+1})$, \mys where \mys $a_{r+1}=0$ \mys and \mys $\forall i\leq r,$\mys$a_i=+1$

							\noindent III. Let $\boldsymbol  {M}$ \mys be a $d\times d$ zero-matrix, $d=r+1$. Populate $M$ as follows:

								\noindent\hspace{20pt}1. Repeat for \mys$i=1$\quad to\quad $i=r-1$:
							\setlength{\belowdisplayskip}{0pt} %\setlength{\belowdisplayshortskip}{0pt}\setlength{\abovedisplayskip}{0pt} \setlength{\abovedisplayshortskip}{0pt}
							%\begin{siderules}
							\begin{alignat*}{7}
										\qquad\qquad & j\mys       & \leftarrow\mys\mys   & \gamma(i)   & \mys\text{and} \quad\mys\mys & k\mys       & \leftarrow\mys\mys   & \gamma(i+1) \\
										\qquad\qquad & m_{jj}\mys  & \leftarrow\mys\mys   & v_j         & \mys\text{and} \quad\mys\mys & m_{jk}\mys  & \leftarrow\mys\mys   & -w_j \\
										\qquad\qquad & w_{k}\mys   & \leftarrow\mys\mys   & w_{k} - w_j & \mys                         &\mys         & \mys                 & \mys
							\end{alignat*}
                            %\end{siderules}
							\noindent\hspace{20pt}2. $j\leftarrow\gamma(r)$, $m_{jj}\leftarrow v_j$, $m_{dj}\leftarrow -w_j$

							\noindent IV. Calculate $\boldsymbol{B}$, $\boldsymbol{D}$ and define $\boldsymbol{f}:\boldsymbol{G}  \rightarrow \{0,1\}$ (Section \ref{sup_NEP_definition}).

							\noindent V. Return $(f(o_1), \dots, f(o_r))$ as KOP's solution vector \hspace{5pt}$\blacksquare$
						}

						\vspace{4pt}\noindent Proof notation follows that in KOP (above) and NEP (Section \ref{sup_NEP_definition}) definitions.

			%\begin{comment}
\subsection{Reverse-Reducing NEP To KOP}\label{reverse_reduction}
			While the KOP-to-NEP reduction proves the later to belong to the same complexity class as
			the former, NEP-to-KOP reduction allows the use of an existing well-known pseudo-polynomial dynamic-programming algorithm  \cite{pisinger_where_2005}
			to solve instances of the former. NEP can be reverse-reduced to KOP by setting $O=G, V=B, W=D$, and \textbf{$c=t$}.

			%\end{comment}
\newpage
\section{Oracle advice on interactions}\label{matrix_OA}
        An Oracle advice (OA) over interactions (edges) rather than genes (nodes), can be represented by a matrix $A=\big [a_{jk}\big ]$ where  $a_{jk}\in\{+1,-1\}$ if $m_{jk}\neq 0$ and $a_{jk}=0$ otherwise (recall $m_{jk}$ is the entry at row $j$ and column $k$ of the interaction matrix $M$, see NEP definition in Section SI \ref{sup_NEP_definition}).  While $m_{jk}\neq 0$ describes what the effect of $g_j$  on $g_k$ actually is, $a_{jk}$ describes what that effect should \textit{ideally} be. A beneficial (damaging) interaction is one where $m_{jk} \times a_{jk} = 1 $ $(m_{jk}\times a_{jk}=-1)$. In other words, an interaction is beneficial (damaging) if it is in agreement (disagreement) with what the Oracle says that interaction should ideally be.  For example, assume $g_j$ inhibits $g_k$, i.e. $m_{jk}=-1$,  but the OA is $a_{jk}=+1$, then $m_{jk}\times a_{jk}=-1$ implies the real effect disagrees with the ideal and the interaction is deemed damaging. The benefit (damage) score of each gene $g_j$, given a matrix OA, is the sum of beneficial (damaging) interactions that $g_j$ is \textit{projecting} onto (out-edges) or \textit{attracting} from (in-edges) other genes in a similar manner as those calculated under a string OA (Section SI \ref{sup_NEP_definition}).

        NEP remains \myC{NP}-hard under a matrix OA. To prove this, we modify the proof in  Section  \ref{KOPdefinition} as follows:


            \vspace{5pt}

            \noindent I. Define $\boldsymbol{\gamma}:\{1,..,r\}\rightarrow \{1,..,r\}$ \mys s.t. $\forall i, 1\leq i <r: w_{\gamma(i)} \leq w_{\gamma(i+1)}$


            \noindent II. Let $\boldsymbol{G}=\boldsymbol{O}+\{o_{r+1}\}$, \mys $\boldsymbol{t}=\boldsymbol{c}$.

            \noindent III. Let $\boldsymbol  {M}$ \mys be a $d\times d$ zero-matrix, $d=r+1$. Populate $M$ as follows:

                    \noindent\hspace{20pt}1. Repeat for \mys$i=1$\quad to\quad $i=r-1$:
                    \setlength{\belowdisplayskip}{0pt} %\setlength{\belowdisplayshortskip}{0pt}\setlength{\abovedisplayskip}{0pt} \setlength{\abovedisplayshortskip}{0pt}
                    %\begin{siderules}
                    \begin{alignat*}{7}
                                \qquad\qquad & j\mys       & \leftarrow\mys\mys   & \gamma(i)   & \mys\text{and} \quad\mys\mys & k\mys       & \leftarrow\mys\mys   & \gamma(i+1) \\
                                \qquad\qquad & m_{jj}\mys  & \leftarrow\mys\mys   & v_j         & \mys\text{and} \quad\mys\mys & m_{jk}\mys  & \leftarrow\mys\mys   & -w_j \\
                                \qquad\qquad & w_{k}\mys   & \leftarrow\mys\mys   & w_{k} - w_j & \mys                         &\mys         & \mys                 & \mys
                    \end{alignat*}
                    %\end{siderules}
                    \noindent\hspace{20pt}2. $j\leftarrow\gamma(r)$, $m_{jj}\leftarrow v_j$, $m_{dj}\leftarrow -w_j$


            \noindent IV. Let  $A$ be a $d\times d$ matrix where:

                        \qquad $\hspace{2cm}a_{jk}  =	\scriptscriptstyle{\begin{cases}	% in math mode, use scriptstyle/scriptscriptstyle, not small/tiny
                                                            +1 & \quad\textrm{if}\quad m_{jk} \neq  0 \\
                                                            0 & \quad\textrm{otherwise}
                                                    \end{cases}
                                                    }$

            \noindent V. Calculate $\boldsymbol{B}$, $\boldsymbol{D}$ as follows:



                    \qquad $b_j = \sum\limits_{k=1}^{n} m_{jk} \oplus a_{jk} \hspace{0.1cm}+\hspace{0.1cm} \sum\limits_{k=1}^{n} m_{kj} \oplus a_{kj} \quad\textrm{where:}$

                    \qquad $\hspace{2cm}m_{xy} \oplus a_{xy}  =	\scriptscriptstyle{\begin{cases}	% in math mode, use scriptstyle/scriptscriptstyle, not small/tiny
                                                        1 & \quad\textrm{if}\quad m_{xy} \times a_{xy} >0 \\
                                                        0 & \quad\textrm{otherwise}
                                                \end{cases}
                                                }$

                    \qquad and similarly the damage score is:

                    \qquad $d_j = \sum\limits_{k=1}^{n} m_{jk} \ominus a_{jk} \hspace{0.1cm}+\hspace{0.1cm} \sum\limits_{k=1}^{n} m_{kj} \ominus a_{kj} \quad\textrm{where:}$

                    \qquad $\hspace{2cm}m_{xy} \ominus a_{xy}  = \scriptscriptstyle{\begin{cases}	% in math mode, use scriptstyle/scriptscriptstyle, not small/tiny
                                                        1 & \quad\textrm{if}\quad m_{xy} \times a_{xy} < 0 \\
                                                        0 & \quad\textrm{otherwise}
                                                \end{cases}
                                                }$


            \noindent VI. Define $\boldsymbol{f}:\boldsymbol{G}  \rightarrow \{0,1\}$ (Section \ref{sup_NEP_definition})


            \noindent  VII. Return $(f(o_1), \dots, f(o_r))$ as KOP's solution vector \hspace{5pt} $\blacksquare$


\newpage
%\setlength{\parskip}{0pt}\setlength{\parsep}{0pt}\setlength{\headsep}{0pt}\setlength{\topskip}{0pt}\setlength{\topmargin}{0pt}\setlength{\topsep}{0pt}\setlength{\partopsep}{0pt}

%\titlespacing{\section}{0pt}{*0}{*0}\titlespacing{\subsection}{0pt}{*0}{*0}\titlespacing{\subsubsection}{0pt}{*0}{*0}
\section{Simulating Evolutionary Pressure}\label{sup_algorithmic_workflow}

%\setlength{\parskip}{0pt}\setlength{\parsep}{0pt}\setlength{\headsep}{0pt}\setlength{\topskip}{0pt}\setlength{\topmargin}{0pt}\setlength{\topsep}{0pt}\setlength{\partopsep}{0pt}

%\titlespacing{\section}{0pt}{*0}{*0}\titlespacing{\subsection}{0pt}{*0}{*0}\titlespacing{\subsubsection}{0pt}{*0}{*0}
%\parbox[t]{\textwidth}{

			The simulation has the parameter tolerance $t$, expressed as percentages of total edges,  indicating the total number damaging interactions to
			be tolerated (equivalently, the knapsack capacity $c$ in the
			corresponding KOP instance).
			For each network, the simulation is carried out under maximum pressure (non-zero OA on every gene)
			against each $t \in {0.1, 1, 5}\%$.
			Given a tolerance value $t$, a knapsack instance is generated from a given NEP instance by reversing the reduction;
			that is: $O=G, V=B, W=D$ and $c=t$. The simulation records the total benefit and damage of objects (=genes, recall $O=G$)
			added to the knapsack by the solver  \cite{pisinger_where_2005} for each round against a randomly generated Oracle advice on each gene.
			The simulation is repeated for 1-5K iterations (sampling threshold, see Section \ref{sup_1Kvs5K}). Figure SI \ref{sup_alg_workflow_fig} summarizes the algorithmic workflow of the simulation.
			%Scrambling of instances is done by randomly distributing the total benefit and damage values in a given instance over all of the genes, in increment of 1. For example, given a total benefits of all genes = $B$, and after all benefit scores are  zeroed out, a benefit of 1 is randomly assigned to a gene $B$ times. The same scrambling scheme is carried out for damages.

%			\begin{comment}

%			\end{comment}
			\begin{figure}[H]
				%\begin{overpic}[width=0.5\textwidth,grid,tics=10]{algorithmic_workflow/algorithmic_workflow.pdf}
				%\begin{overpic}[width=16.5cm, height=15.5cm, right]{algorithmic_workflow/algorithmic_workflow_cropped.pdf}
				%\begin{overpic}[scale=.7, right]{01.algorithmic_workflow/algorithmic_workflow_cropped3.pdf}
				\begin{overpic}[width=.9\textwidth, height=13cm, right]{01.algorithmic_workflow/PNAS/combined-flow-and-runtime.png}
					\put (0,51) {
						\parbox[l]{.57\textwidth}{
							\scriptsize{% \tiny, \scriptsize, \footnotesize, \small, \normalsize, \large, \Large, \LARGE, \huge, and \Huge.
					            \textbf{right}: Simulations are carried at a certain pressure. Maximum pressure is when the Oracle has a non-zero advice on all nodes. Some simulations were carried at lower pressure levels where the Oracle is indifferent to 25, 50, 0r 75\% of genes.
					            For each tolerance $t$ value, 1-5K simulation rounds are carried out. In each round,
					            a random OA is generated on all genes (nodes), followed by a calculation of benefit/damage
					            value for each node against the current OA. The resulting NEP instance is reverse-reduced
					            to a KOP instance ($O=G_i, V=B_i, W=D_i, c=t_i$) and fed to a knapsack solver.
								In each round, the sequences $G_i, B_i, D_i, t_i,$ and $S_i$ are written to file, where $S_i$ is the solution vector
								$(s_1, \dots, s_k), k=|G_i|,$ and $s_i\in\{0,1\}$. $s_i=1$ ($s_i=0)$ implies "conserve" ("delete") or, in the context of the knapsack problem, "inside" ("outside") the knapsack.
								\newline
								\textbf{below}:	average algorithm running time in milliseconds for each network.
								'S' denotes an identical simulation on a second computer cluster different from the first run. For $t$=0.1\%, the execution times are too negligible as a result of the  dynamic programming algorithm  \cite{pisinger_where_2005}  being upper-bounded by an exponent  = $O(c)$ value. We therefore carried out the simulation at higher tolerance values $t\in\{5, 25, 50\} \%$. NL has significantly less nodes compared to other networks, and therefore shows the smallest execution times. PPI, RN and NH have ${\sim}$equal network sizes, but instances in PPI are solved faster compared to to its smaller instance sizes (a majority of genes being having either benefit (damage) as zero, and therefore such genes are not part of the optimization search as they should be conserved (deleted) regardless, see discussion on effective instance size (EIS) in the main text for details).
								}
						}
					}
				\end{overpic}
				\caption{The algorithmic workflow of computer simulation and the average run time of the knapsack solver.}
				\label{sup_alg_workflow_fig}
			\end{figure}

\newpage
\section{Effect of Sampling Threshold}\label{sup_1Kvs5K}
Increasing the sampling threshold in the simulation (i.e. how many NEP instances to simulate) does not change the results, due to the effect of the Central Limit Theorem  \cite{kallenberg_foundations_2006}. Figures SI \ref{supp_1Kvs5K_corr} and \ref{supp_1Kvs5K_wheel} %and \ref{supp_1Kvs5K_EGB}
compare the results computed over 1,000 versus 5, 000 simulated instances (see the corresponding Figures 4 and 5 in the main text for detailed description).
\subsection{Benefit-Damage Correlation:}

\begin{figure}[H]
		\centering
				\includegraphics[scale=.6]{/10.1K.vs.5K/scatter/processed.png}
				\caption{Increasing the sampling threshold from (A) 1,000 to (B) 5,000 NEP has virtually no effect on the resulting benefit-damage correlations.}
				\label{supp_1Kvs5K_corr}
\end{figure}

\subsection{Effective Instance Size:}

\begin{figure}[H]
		\centering
				\includegraphics[scale=.6]{/10.1K.vs.5K/wheel/processed.png}
				\caption{Increasing the sampling threshold from (A) 1,000 to (B) 5,000 NEP has minimal to no effect on effective instance size (EIS). Legend: numbers between parenthesis are average +/- standard deviation. }
				\label{supp_1Kvs5K_wheel}
\end{figure}

\begin{comment}
    \subsection{Gained Benefits:}
    \begin{figure}[H]
    		\centering
    				\includegraphics[scale=.6]{/10.1K.vs.5K/ETX/processed.png}
    				\caption{Increasing the sampling threshold from (A) 1,000 to (B) 5,000 NEP has minimal to no effect on Gained Benefits (GB).}
    				\label{supp_1Kvs5K_EGB}
    \end{figure}
\end{comment}
\newpage
\section{Benefit-damage correlation}\label{sup_extra_BD_corr}
    Figures SI \ref{fig_data_stress_Regulatory} and \ref{fig_data_stress_DBs} show the benefit-damage correlation results for the regulatory and DB-sourced networks, respectively. For detailed description please see Figure 4 in the main text which shows the results for PPI networks.

    \begin{figure}[H]%[!htb]
        \includegraphics[width=1.0\textwidth]{04.scatter/Thesis/ch3/data_stress/Regulatory/processed_Regulatory.png}
        \caption
                {
                    benefit-damage correlation in regulatory networks.
                }
        \label{fig_data_stress_Regulatory}
    \end{figure}

    \begin{figure}[H]%[!htb]
                \includegraphics[width=1.0\textwidth]{04.scatter/Thesis/ch3/data_stress/DB-sourced/processed_DBsourced.png}
                \caption
                        {
                            benefit-damage correlation in database-sourced networks.
                        }
                \label{fig_data_stress_DBs}
            \end{figure}
\newpage
\section{Predicted vs. actual degree distributions}\label{sup_prediction}

        Figures SI \ref{fig:predictions_PPIs}, \ref{fig:predictions_Regulatory} and \ref{fig:predictions_DBsourced} show detailed plots of the actual versus predicted degree distribution of PPI, regulatory and DB-sourced networks, respectively, along with detailed bar plots of each $(\alpha,\beta)$ values used in the prediction formula and their respective proportionality to the node:edge $(n2e)$
        and edge:node $(e2n)$ ratios in each networks.
        The $(\alpha,\beta)$ values were  numerically determined by considering each $\alpha$ in the interval [0.01, 1] in increments of 0.01 against each $\beta$ in [0.1, 10] interval in increments of 0.1.
        %The $(\alpha,\beta)$ that produced the best prediction correlates closely with $(n2e,e2n)$ for most networks as seen in the bar plots in Figures SI \ref{fig:predictions_PPIs}, \ref{fig:predictions_Regulatory} and \ref{fig:predictions_DBsourced}.
        Hub prediction may visually appear  to be less precise but that is only due to the log scale in the y-axis. High discrepancies between $(\alpha,\beta)$ and $(n2e,e2n)$ values can be used to infer the quality of coverage and resolution of a network, and the extend to which it represents a representative sample the overall true and complete network. For example, $e2n >> \beta$ for the Yeast BioGrid network (Figure SI \ref{fig:predictions_DBsourced}, right bar plot). Examining the degree distribution of this network (Figure SI \ref{fig:deg_dist_DB}), the frequency of degree-1 nodes is significantly low (${\sim}19\%$) compared to all other networks (DB-sourced, regulatory or PPI networks, where degree-1 frequency is 44$\pm$10\%). The Worm BioGrid network on the other hand, has $\beta >> e2n$, which can be explained by the under representation of hub nodes in its network (it has no genes of degree $\geq$ 9, while on average 8$\pm$5\% of genes in other networks have degree $\geq$9).
        %In a future work, we are using the $(n2e,e2n)$ values of the highly resolute Human Iso network reported by Yang et \textit{al.}  \cite{yang_widespread_2016} as gold-standard $(\alpha,\beta)$ values in order to estimate the quality of coverage of resolution of other networks (further discussed in Section \ref{sec:conclusion_applications}).
        %

    	\begin{figure}[H]%[!htb]
    			\centering
    					\includegraphics[width=1.0\textwidth]{08.deg_dist_prediction/Thesis/ch3/predictions/PPIs/processed/combined_predictions_alpha_beta_n2e_e2n.png}
    					\caption
                            [
                                Actual and predicted degree distribution of PPI networks.
                            ]
                            {
                                Actual and predicted degree distribution of PPI networks. The bar plots (bottom) show the $\alpha$ and $\beta$ values in the predicted networks versus the node:edge ($n2e$) and edge:node ($e2n$) ratios of the real networks.
                            }
    					\label{fig:predictions_PPIs}
    	\end{figure}

%\subsection{regulatory networks}
        \begin{figure}[H]%[!htb]
                \centering
                        \includegraphics[width=1.0\textwidth]{08.deg_dist_prediction/Thesis/ch3/predictions/Regulatory/processed/combined_predictions_alpha_beta_n2e_e2n.png}
                        \caption
                            [
                                Actual and predicted degree distribution of regulatory networks.
                            ]
                            {
                                Actual and predicted degree distribution of regulatory networks. The bar plots (bottom) show the $\alpha$ and $\beta$ values in the predicted networks versus the node:edge ($n2e$) and edge:node ($e2n$) ratios of the real networks.
                            }
                        \label{fig:predictions_Regulatory}
        \end{figure}

%\subsection{database-souced networks}
        \begin{figure}[H]%[!htb]
                \centering
                        \includegraphics[width=1.0\textwidth]{08.deg_dist_prediction/Thesis/ch3/predictions/DBsourced/processed/combined_predictions_alpha_beta_n2e_e2n.png}
                        \caption
                        [
                            Actual and predicted degree distribution of database-sourced networks.
                        ]
                        {
                            Actual and predicted degree distribution of database-sourced networks. The bar plots (bottom) show the $\alpha$ and $\beta$ values in the predicted networks versus the node:edge ($n2e$) and edge:node ($e2n$) ratios of the real networks.
                        }
                        \label{fig:predictions_DBsourced}
        \end{figure}

\newpage
\section{Simulated evolution}\label{sup_sim_evo}
    Figure SI \ref{sup-workflow} shows the algorithmic workflow of the simulated evolution. In the fitness calculation step (denoted by a star in Figure SI  \ref{sup-workflow}), the fitness of a network vis-a-vis the current NEP instance is calculated based on 1) effective instance size (EIS) and 2) effective gained benefits (EGB) \cite{atiia_computational_2017-1}.
    EIS is defined as the \% of unambiguous nodes in the instance:


        \begin{align}
            \qquad\qquad\qquad\qquad\qquad\qquad EIS=\frac{|\{n_i:b_i=0|d_i=0\}|}{N}
        \end{align}


    \vspace{3mm}\noindent where $N$ is the total number of nodes and $b_i$ and $d_i$ and the benefit and damage score of some node $n_i$. Let $(s_1, s_2, .., s_k)$ be the solution vector to an NEP instance where $s_i\in\{0,1\}$  and $s_i=1 \mys (s_i=0)$ implies "conserve" ("delete"). The multiset $B=\{b_i: s_i=1\}$ contains a list of all the benefits of nodes that are to be optimally conserved, and the  effective gained benefits EGB=$sum(set(B))$.  EGB is hence $B$ normalized by the number of nodes it takes  to add a certain benefit value). For example, with
    $B1=\{1,1,1,5\}$
    and
    $B2=\{2,6\}$, $sum(B1)=sum(B2)=8$.
    But
    $EGB1 = sum(set(B1))=sum({1,5})= 6$
    while
    $EGB2 = sum(set(B2))=sum({2,6})=8$.
    Let $B_{tot}$ be the total benefit in a given NEP instance (the sum of gained benefits of conserved genes and lost benefits of deleted genes), the fitness of a given NEP instance $S$ is measured as:

        \begin{align}
            \qquad\qquad\qquad\qquad\qquad\qquad F(S) = EIS^\alpha \times \frac {EGB}{B_{tot}}
        \end{align}



    \vspace{3mm}\noindent where $\alpha\in \approx \mathbb{R}^{+}$. We applied $\alpha=2$ in all simulation. With  $\alpha$, the weight  given to EIS vs EGB can be calibrated. This reflects the inherent opposition of EIS vs EGB, as EIS is best minimized with a large number of leaves while EGB is maximized with a large number of hubs. While EGB indicates how well a network accumulates as many beneficial interactions as possible with the smallest possible number of genes to conserve, normalizing it by the total benefits $B_{tot}$ penalizes networks that hemorrhage beneficial interactions that are lost to deleted genes in the optimal solution. The threshold $t$ of tolerated damaging interactions in the solution is imposed at 5\% of the sum of all damages in all simulations.


    Figures SI \ref{fig:sim_evo_ppi}, \ref{fig:sim_evo_reg} and \ref{fig:sim_evo_db} show extended results of the degree distribution of synthetically evolved networks grouped by network families (PPIs, regulatory and DB-sourced networks respectively). In each simulation, the network growth (add-edge and add-node mutations) are halted when the size of the synthetic network has become equal to the corresponding real network. Once network growth is disabled, the evolutionary algorithm is further run for a constant 2000 generations with re-assign edge mutation only.


    \begin{figure}[H]
        \centering
        \includegraphics[scale=.42]{11.ACM-BCB/Science/workflow_transparent_2nd_draft.png} % original width= 42.65cm,  height = 19.86cm
                \caption{The algorithmic workflow of the evolutionary algorithm. Simulations begin with empty networks or seed networks that have randomly distributed edges. Each network is randomly mutated by reassigning one edge at each generation and, if growth is allowed, one node is also added along with as many randomly assigned edges as needed to maintain the desired edge:node ratio. An instance of the network evolution problem (NEP) is obtained by generating a random Oracle advice (OA) on all edges in the network. A network's fitness at each instance $S$ is calculated following the $F(S)$ formula (see text). The 10\% of networks with the highest average fitness over all instances are selected to breed a population of networks for the subsequent generation. Adapted with modification with permission from  \cite{atiia_computational_2017-1}.
                }
        \label{sup-workflow}
     \end{figure}





    	\begin{figure}[H]%[!htb]
    			\centering
    					\includegraphics[width=1.0\textwidth]{11.ACM-BCB/Science/v2/src/plot_groups/png/PPIs.png}
    					\caption
                            {
                                Evolving synthetic networks to the same size (number of nodes and edges) as protein-protein interaction networks.
                            }
    					\label{fig:sim_evo_ppi}
    	\end{figure}

    	\begin{figure}[H]%[!htb]
    			\centering
    					\includegraphics[width=1.0\textwidth]{11.ACM-BCB/Science/v2/src/plot_groups/png/Regulatory.png}
    					\caption
                            {
                                Evolving synthetic networks to the same size (number of nodes and edges)  as regulatory networks.
                            }
    					\label{fig:sim_evo_reg}
    	\end{figure}

    	\begin{figure}[H]%[!htb]
    			\centering
    					\includegraphics[width=1.0\textwidth]{11.ACM-BCB/Science/v2/src/plot_groups/png/DBSourced.png}
    					\caption
                            {
                                Evolving synthetic networks to the same size (number of nodes and edges)  as database-sourced networks.
                            }
    					\label{fig:sim_evo_db}
    	\end{figure}
\clearpage %https://tex.stackexchange.com/questions/2958/why-is-newpage-ignored-sometimes
\newpage
\section{Simulated adaptation}\label{sup_sim_adapt}
        In these experiments, the simulated network evolution algorithm (see Figure SI \ref{sup-workflow}) starts with networks that have a number of nodes/edges equal that of a corresponding real MIN. The edges of the seed networks are initially randomly assigned. In each generation, only reassign-edge mutation is carried out (no add-node or add-edge mutations) as opposed to the simulated evolution experiments (Section SI \ref{sup_sim_evo}). Figures SI \ref{fig:sim_evo_adapt_4x} and \ref{fig:sim_evo_adapt_8x} show the degree distribution of the fittest synthetic network (labelled `Simulation') against that of the corresponding equal-size MIN after 4X and 8X generations of mutate-and-select, respectively, where  X equals the number of the nodes in the network. The degree distribution of the initial seed network is labeled 'Seed' in Figures SI \ref{fig:sim_evo_adapt_4x} and \ref{fig:sim_evo_adapt_8x}. The threshold $t$ of tolerated damaging interactions in the solution is kept at 5\% of the sum of all damages in all simulations, as was the case in the simulated evolution experiments of Section \ref{sup_sim_evo}.

    	\begin{figure}[H]%[!htb]
    			\centering
    					\includegraphics[width=1.0\textwidth]{11.ACM-BCB/Science/v2/src/plot_groups/png/Adapt_simulations_x4_with_seed.png}
    					\caption
                            {
                                Adapting synthetic networks for 4X generations of mutate-and-select where X is the total number of nodes in the networks. Simulation starts with the a network that has the same number of nodes and edges as the corresponding real MIN, but with edges randomly assigned to nodes. The  degree distribution of the synthetic (Simulation) shown here is that of the fittest network after 4X generation of mutate-only simulated adaptation. Increasing the number of generations does not significantly change the degree distribution (see Figure SI \ref{fig:sim_evo_adapt_8x}). The degree distribution of the initial seed network is labeled 'Seed'.
                            }
    					\label{fig:sim_evo_adapt_4x}
    	\end{figure}

\begin{figure}[H]%[!htb]
        \centering
                \includegraphics[width=1.0\textwidth]{11.ACM-BCB/Science/v2/src/plot_groups/png/Adapt_simulations_x4_with_seed.png}
                \caption
                    {
                        The same simulation as that in Figure SI \ref{fig:sim_evo_adapt_4x} except that here the simulation is further continued for 4X more generations (hence the total number of generations is 8X the number of nodes in the network). Such increase in number of generations does not significantly change the degree distribution as opposed to simulations terminated after 4X generations (Figure SI \ref{fig:sim_evo_adapt_4x}). The degree distribution of the initial seed network is labeled 'Seed'.
                    }
                \label{fig:sim_evo_adapt_8x}
\end{figure}

%\newpage
\clearpage %https://tex.stackexchange.com/questions/2958/why-is-newpage-ignored-sometimes
%\noindent \large{\textbf{References:}}
%\noindent See main text's Reference section

%when biblatex
%\printbibliography

\newpage
%when bibtex
\bibliography{/Volumes/DataHub/Publishing/z-misc/zotero-library/better_bibtex_standalone}
\bibliographystyle{vancouver}

\end{document}

$A =(a_1,a_2,a_3,a_4,a_5,a_6,a_7)$

$\hspace{4.2mm}=(0,+1,-1,+1,0,0,-1)$

\vspace{.3cm}

{\Large \textcircled{ }} {\Large \textcircled{ }} {\Large \textcircled{ }}

\vspace{.2cm}

{\Large \textcircled{ }} {\Large \textcircled{ }} {\Large \textcircled{ }} {\Large \textcircled{ }}

\vspace{.2cm}

\normalsize $\mathnormal{g_1}$ \normalsize $\mathnormal{g_2}$ \normalsize $\mathnormal{g_3}$ \normalsize $\mathnormal{g_4}$ \normalsize $\mathnormal{g_5}$ \normalsize $\mathnormal{g_6}$ \normalsize $\mathnormal{g_7}$

\ding{1}




